### ISSUE TITLE
Homey stops listening after "Homey starts listening" card

### ISSUE THREAD

**Issue by novaflash** [athombv/homey#468]
**What is your Homey version (Settings → System)?**
0.8.30

**What did you try to achieve?**
I wanted Homey to say "What?" when I say "OK Homey", so that I can hear that Homey is paying attention to me. Then I can say what I want to say and Homey can act on that. I tried to do this by creating a new flow with the "Homey starts listening" card, with the action "Say something" and entering the word "What?".

**What was the result?**
When I say "OK, Homey", Homey responds correctly with "What?", but then quits listening.

**What did you expect as result?**
I expect Homey to listen to what I had to say.

**Any other remarks**
Perhaps an action card in the category "Microphone" could be added that is titled "Start listening" (and stop listening), so I can add that as a secondary action. Obviously this should not trigger the same flow itself, or it would become a recursive loop.

**Comment by Rubelke** [comment]
I'm having the same issue at 0.8.30
When using a start listening card homey does not wait for the answer YES/No
But when you use the test flow command everything works OK.
So it seems that there is a difference between OK homey start flow .... and the TEST command in the flow editor.

**Comment by WeeJeWel** [comment]
When Homey talks, it disables the microphone because it interferes. For now I'll close this issue (as it's intended behavior), maybe we'll add support this later.

**Comment by novaflash** [comment]
Sorry, but, this doesn't make sense. I understand that while Homey is actively talking, that Homey should not be listening, as this will confuse the microphone. This is a no-brainer. But after it's done talking, it should be listening for a command I want to give Homey.

Currently when I do this:
When: You say something
Then: Say something
...it effectively breaks all voice input because Homey will now never listen to any of the commands I give. It will listen to "Ok Homey" and then it will say something and then it will stop listening. This breaks voice control completely; with the simple flow that I have created I have broken Homey's voice control.

I argue that it should resume listening to my command after it has said something.

You may disagree with this, of course, and I totally understand that you guys are in control of the product development direction, but this is so wrong, that I have to oppose this.

I also understand that this might not be top priority, but I don't believe it should just be closed. This is a flaw in the product.

However, if there is another way that I can let the Homey let me know with an audio cue that it is currently listening to a command from me, that the "OK Homey" trigger was successfully detected and it is now waiting for input, without having to actually look at the Homey, then please, let me know how this can be done. Otherwise, my points above stand.

**Comment by WeeJeWel** [comment]
I understand your point :)

In the meantime, you can use https://apps.athom.com/app/com.athom.soundboard to play an audible notification.

**Comment by novaflash** [comment]
Okay, I'll try that; at least it's something.

**Comment by novaflash** [comment]
The soundboard option is a bust. I have installed soundboard, and uploaded a .wav file. I have made a new flow:

When: Homey starts listening
Then: Soundboard, play a sound (my uploaded file)

When I say "OK Homey", it simultaneously listens to input, and it plays the soundboard audio file. The audio it plays is fed into the microphone and as soon as the file is done playing, Homey stops listening and processes the audio it just played back as speech, which obviously fails horribly.

**Comment by WeeJeWel** [comment]
Depends on your sound. A 'blib' works fine in my experience, but speech will interfere of course :)

**Comment by novaflash** [comment]
Wouldn't it make a lot more sense if Homey's listening was suspended while the wav file was playing, and only THEN start listening? Just as how suspending Homey's listening while it was saying something, and only after it has finished saying something, start listening, would make a lot more sense? The only option I have now is the shortest and tiniest of blip sounds, or it will do exactly what you already thought would happen; it would listen to itself and interfere with speech recognition.

I will choose to believe that you are closing this issue because you have more pressing things to do, and will come back to fix this issue later. I will choose to believe that you are not closing this ticket because I want something out of the ordinary.

**Comment by novaflash** [comment]
One final thing; I am using a wav file of 0.28 seconds and it STILL fails.

**Comment by novaflash** [comment]
I lied; one more final final thing. If I do a very silent wav file, I can still use the Homey's voice control. It seems like the automatic gain on the microphone of the Homey gets thrown off a bit when I use a "normal" volume in the wav file (even if it's only 0.2 seconds long!) and then somehow thinks it's done listening and it doesn't hear anything new at a sufficient sound level.

If I get closer and talk loudly, it works, with both a normal and low volume wav file.
If I move further away and use a very low volume wav file, it works. But with a normal volume wav file it stops listening as it seems to not hear my low volume voice.

In short; audio pick-up should not be on while a wav file or text-to-speech output is running. Audio picked-up should resume directly after it has finished playing back the wav file or text-to-speech.
